### PR TITLE
Bump to Rector ~2.0.0

### PR DESCRIPTION
**Pull request by samsonasik**
| Q             | A
| ------------- | ---
| QA?  | ✔️

Bump to just released Rector 2.0.0 https://github.com/rectorphp/rector/releases/tag/2.0.0

### CODE REVIEW

**Comment by samsonasik** [comment]
@roxblnfk ready to merge 👍

File: src/Core/tests/Scope/UseCaseTest.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function testScopeBindingsAsNotSingletons(bool $theSame, string $alias, m
 
     public static function provideScopeBindingsAsNotSingletons(): iterable
     {
-        yield 'array-factory' => [false, 'foo', [Factory::class, 'makeStdClass']];
+        yield 'array-factory' => [false, 'foo', (new Factory())->makeStdClass(...)];
```
Comment (roxblnfk) [inline src/Core/tests/Scope/UseCaseTest.php:82]: Need to revert it because we test callable arrays in this test case

Comment (samsonasik) [reply src/Core/tests/Scope/UseCaseTest.php]: reverted https://github.com/spiral/framework/pull/1177/commits/38ee09a40f18f496be2d2e5da6d00ecbbfe414bf 👍

**Review by roxblnfk — APPROVED** [review]

**Comment by roxblnfk** [comment]
Thanks!

**Pull request merged**